frontend/src/store/middlewares/socket/chat-socket.middleware.ts

```diff
@@ -0,0 +1,44 @@
+import { Middleware } from '@reduxjs/toolkit';
+import { SocketEvent, SocketNamespace } from 'common/enums/enums';
+import {
+  AppDispatch,
+  ChatMessageGetAllItemResponseDto,
+} from 'common/types/types';
+import { io } from 'socket.io-client';
+import { chatsActions } from 'store/actions';
+
+type SocketMiddlewareParams = {
+  dispatch: AppDispatch;
+};
+
+const chatSocketInstance = io(SocketNamespace.CHAT);
+
+const chatSocket: Middleware =
+  ({ dispatch }: SocketMiddlewareParams) =>
+  (next) =>
+  (action): void => {
+    if (chatsActions.joinRoom.match(action)) {
+      chatSocketInstance.emit(SocketEvent.CHAT_JOIN_ROOM, action.payload);
+      chatSocketInstance.on(
+        SocketEvent.CHAT_ADD_MESSAGE,
+        (message: ChatMessageGetAllItemResponseDto): void => {
+          dispatch(chatsActions.addMessage(message));
+          // PR Read Messages should be Merged
+          // dispatch(chatsActions.readMessages(message.chatId));
```

Comment:
Left a comment here, because it's something easy to forget. Can be changed, after the specified PR is merged

Reply:
Start the comment with TODO:

Reply:
it will be easy to find such comments later